docs/project-model/project-json-reference.md

```diff
@@ -0,0 +1,847 @@
+
+<a name="resourceBuiltIn"></a>
+## resourceBuiltIn
+Type: String or String[] with a globbing pattern
+
+Specifies resource files for the project.
+
+For example:
+
+    {
+        "resourceBuiltIn": [ "compiler/resources/**/*", "**/*.resx" ]
+    }
+
+<a name="excludeBuiltIn"></a>
+## excludeBuiltIn
+Type: String or String[] with a globbing pattern
+
+Specifies what files to exclude from the build.
+
+For example:
+
+    {
+        "excludeBuiltIn": ["bin/**", "obj/**"]
+    }
+
+<a name="dependencies"></a>
+## dependencies
+Type: Object
+
+An object that defines the package dependencies of the project, each key of this object is the name of a package and each value contains versioning information.
+
+For example:
+
+    "dependencies": {
+        "System.Reflection.Metadata": "1.2.0-rc3-23811",
+        "Microsoft.Extensions.JsonParser.Sources": {
+          "type": "build",
+          "version": "1.0.0-rc2-16453"
+        },
+        "Microsoft.Extensions.HashCodeCombiner.Sources": {
+          "type": "build",
+          "version": "1.0.0-rc2-16054"
+        },
+        "Microsoft.Extensions.DependencyModel": "1.0.0-*"
+    }
+
+<a name="tools"></a>
+## tools
+Type: Object
+
+An object that defines package dependencies that are used as tools for the current project, not as references. Packages defined here are available in scripts that run during the build process, but they are not accessible to the code in the project itself. Tools can for example include code generators or post-build tools that perform tasks related to packing.
+
+For example:
+
+    {
+        "tools": {
+            "MyObfuscator": "1.2.4"
+        }
+    }
+
+<a name="commands"></a>
+## commands
+Type: Object
+
+TODO: Investigate actual status
+
+Commands are deprecated in the cli. (?)
+
+<a name="scripts"></a>
+## scripts
+Type: Object
+
+An object that defines scripts run during the build proces. each key in this object identifies where in the build the script is run, each value is either a string with the script to run or an array of strings containg scripts that will run in order.
+The supported events are:
+* precompile
+* postcompile
+* prepublish
+* postpublish
+
+For example:
+
+    {
+        "scripts": {
+            "precompile": "generateCode.cmd"
+            "postcompile": [ "obfuscate.cmd", "removeTempFiles.cmd" ]
+        }
+    }
+
+<a name="compilationOptions"></a>
+## compilationOptions
+Type: Object
+
+An object whose properties control various aspects of compilation, the valid properties are listed below.
+
+For example:
+
+    "compilationOptions": {
+      "allowUnsafe": true
+    }
+
+<a name="define"></a>
+### define
+Type: String[]
+
+A list of defines such as "DEBUG" or "TRACE" that can be used in conditional compilation in the code.
+
+For example:
+
+    {
+        "compilationOptions": {
+            "define": ["TEST", "OTHERCONDITION"]
+        }
+    }
+
+<a name="nowarn"></a>
+### nowarn
+Type: String[]
+
+A list of warnings to ignore.
```

Comment:
What list of warnings?  The example seems to indicate that warnings aren't output when you compile.